packages/fs-detectors/src/detect-framework.ts

```diff
@@ -90,3 +95,18 @@ export async function detectFramework({
   );
   return result.find(res => res !== null) ?? null;
 }
+
+export async function detectFrameworkRecord({
+  fs,
+  frameworkList,
+}: DetectFrameworkRecordOptions): Promise<Framework | null> {
+  const result = await Promise.all(
+    frameworkList.map(async frameworkMatch => {
+      if (await matches(fs, frameworkMatch)) {
+        return frameworkMatch;
+      }
+      return null;
+    })
+  );
+  return result.find(res => res !== null) ?? null;
+}
```

Comment:
We could probably make `detectFramework` use `detectFrameworkRecord`, but I was having trouble reconciling the need for different types between them. It's `BaseFramework` vs. `Framework`.

Reply:
I'm thinking we can completely drop `detectFramework()` in favor of `detectFrameworkRecord` but that can be a future PR since it will be a major semver.

Might be nice to add `// TODO` for this so the next person reading knows our intentions 😃

Reply:
Updated in https://github.com/vercel/vercel/pull/9009#:~:text=Resolve%20conversation-,PR%20feedback,-fbf562e